matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/space/DefaultSpaceService.kt

```diff
@@ -127,24 +127,29 @@ internal class DefaultSpaceService @Inject constructor(
                     ),
                     second = response.rooms
                             ?.filter { it.roomId != spaceId }
-                            ?.map { childSummary ->
-                                val childStateEv = response.events
-                                        ?.firstOrNull { it.stateKey == childSummary.roomId && it.type == EventType.STATE_SPACE_CHILD }
-                                val childStateEvContent = childStateEv?.content.toModel<SpaceChildContent>()
-                                SpaceChildInfo(
-                                        childRoomId = childSummary.roomId,
-                                        isKnown = true,
-                                        roomType = childSummary.roomType,
-                                        name = childSummary.name,
-                                        topic = childSummary.topic,
-                                        avatarUrl = childSummary.avatarUrl,
-                                        order = childStateEvContent?.order,
-                                        autoJoin = childStateEvContent?.autoJoin ?: false,
-                                        viaServers = childStateEvContent?.via ?: emptyList(),
-                                        activeMemberCount = childSummary.numJoinedMembers,
-                                        parentRoomId = childStateEv?.roomId
-                                )
-                            }.orEmpty()
+                            ?.flatMap { childSummary ->
+                                response.events
+                                        ?.filter { it.stateKey == childSummary.roomId && it.type == EventType.STATE_SPACE_CHILD }
+                                        ?.map { childStateEv ->
+                                            // create a child entry for everytime this room is the child of a space
+                                            // beware that a room could appear then twice in this list
+                                            val childStateEvContent = childStateEv.content.toModel<SpaceChildContent>()
```

Comment:
Return null if `childStateEvContent` is null?

Reply:
ok